crypto/ec/curve448/field.h

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2017-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2014 Cryptography Research, Inc.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Originally written by Mike Hamburg
+ */
+
+#ifndef HEADER_FIELD_H
+# define HEADER_FIELD_H
+
+# include "internal/constant_time_locl.h"
+# include <string.h>
+# include <assert.h>
+# include "word.h"
+
+# define NLIMBS (64/sizeof(word_t))
+# define X_SER_BYTES 56
+# define SER_BYTES 56
+
+# if defined(__GNUC__) || defined(__clang__)
+#  define INLINE_UNUSED __inline__ __attribute__((__unused__,__always_inline__))
+#  define RESTRICT __restrict__
+#  define ALIGNED __attribute__((aligned(32)))
```

Comment:
underscores could be used for `__aligned__` as well.

Reply:
Note however that based on the comment from @bitwiseshiftleft
the alignment here is used to match the alignment of a vector type
that is aliased with the gf_s struct at another place.
That is ok (from my GCC perspective), if the aliased vector is
a vector of word_t, thus the so called "alias set" of word_t and
vector of word_t is identical.
It could of course be more portable by placing all possible aliases
in a union inside gf_s, then the alignment would be automatically
correct.

Reply:
I changed to use underscores. However I just removed the vector type code anyway (see below), so this alignment might be excessive. I've left it for now in case we resurrect the vector code later.